controller/k8s/api.go

```diff
@@ -1066,6 +1066,26 @@ func (api *API) GetServiceProfileFor(svc *corev1.Service, clientNs, clusterDomai
 	}
 }
 
+// GetNsAnnotationFor gets the value of an annotation for a namespace.
+func (api *API) GetNsAnnotationFor(name string, annotation string) (string, error) {
+	obj, err := api.GetObjects("", k8s.Namespace, name, labels.Everything())
```

Comment:
Why use `GetObjects` instead of just `api.NS().Lister().Get`? This avoids needing to cast.